app/models/user.rb

```diff
@@ -72,12 +73,16 @@ class User < ApplicationRecord
   validate :toxic_email_domain, on: :create
 
   def self.authenticate(who, password)
-    user = find_by(email: who.downcase) || find_by(handle: who)
+    user = find_by_email(who) || find_by(handle: who)
     user if user&.authenticated?(password)
   rescue BCrypt::Errors::InvalidHash
     nil
   end
 
+  def self.find_by_email(email)
+    find_by("lower(email) = ?", email.downcase)
```

Comment:
should we instead do `lower(email) = lower(?)`, so we have consistent downcasing in case it differs between postgres & ruby?

Reply:
ok, let's do that, good idea

Reply:
```suggestion
    find_by("lower(email) = lower(?)", email)
```